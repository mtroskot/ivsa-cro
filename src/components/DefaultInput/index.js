import React from 'react';
import PropTypes from 'prop-types';
import { TextInput } from 'react-native';
import styles from 'src/components/DefaultInput/styles';

const DefaultInput = props => {
  const { onSubmitEditing, textInputRef } = props;
  // TODO refactor spread props
  return (
    <TextInput
      underlineColorAndroid="transparent"
      ref={textInputRef}
      onSubmitEditing={onSubmitEditing}
      {...props}
      style={[styles.default, props.style]}
    />
  );
};

DefaultInput.propTypes = {
  onSubmitEditing: PropTypes.func,
  textInputRef: PropTypes.shape({ current: PropTypes.elementType }),
};

export default React.memo(DefaultInput);
