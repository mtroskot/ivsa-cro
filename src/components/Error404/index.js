import React from 'react';
import { Image, Text, View } from 'react-native';
import sadFace from 'src/assets/images/emoji/sadFace.png';
import styles from 'src/components/Error404/styles';

const Error404 = () => {
  return (
    <View style={styles.container}>
      <Image source={sadFace} style={styles.image} />
      <Text style={styles.errorText}>Oops Something went wrong</Text>
    </View>
  );
};

export default React.memo(Error404);
