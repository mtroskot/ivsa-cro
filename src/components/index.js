export { default as CustomButton } from 'src/components/CustomButton';
export { default as Error404 } from 'src/components/Error404';
export { default as FloatingLabelTextInput } from 'src/components/FloatingLabelTextInput';
export { default as IconButton } from 'src/components/IconButton';
export { default as PopupMessage } from 'src/components/PopupMessage';
export { default as KeyboardAvoidAndDismissView } from 'src/components/KeyboardAvoidAndDismissView';
export { default as ListItem } from 'src/components/ListItem';
export { default as Loader } from 'src/components/Loader';
export { default as PagingButtons } from 'src/components/PagingButtons';
