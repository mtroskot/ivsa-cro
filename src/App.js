import React from 'react';
import { Provider } from 'react-redux';
import { StyleSheet } from 'react-native';
import { SafeAreaProvider } from 'react-native-safe-area-context';
import SafeAreaView from 'react-native-safe-area-view';
import { PersistGate } from 'redux-persist/lib/integration/react';
import AppContainer from 'src/navigation';
import { NavigationService } from 'src/services';
import store, { getPersistor } from 'src/store';
import { PopupMessage } from 'src/components';

const persistor = getPersistor();

export default function App() {
  return (
    <Provider store={store}>
      <SafeAreaProvider>
        <SafeAreaView style={styles.container} forceInset={{ top: 'never' }}>
          <PersistGate loading={null} persistor={persistor} bootstrapped={true}>
            <AppContainer ref={ref => NavigationService.setTopLevelNavigator(ref)} />
            <PopupMessage />
          </PersistGate>
        </SafeAreaView>
      </SafeAreaProvider>
    </Provider>
  );
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    backgroundColor: '#fff'
  }
});
