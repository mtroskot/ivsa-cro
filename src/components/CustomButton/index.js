import React from 'react';
import PropTypes from 'prop-types';
import { Text, TouchableOpacity, View, ViewPropTypes } from 'react-native';
import Icon from 'react-native-vector-icons/Ionicons';
import { dimensions } from 'src/styles';
import { AppUtils } from 'src/utils';

const { rem } = dimensions;

const CustomButton = props => {
  const { iconProps, tOpacityStyle, viewStyle, textStyle, iconStyle, text, onPress } = props;
  const buttonText = text ? <Text style={textStyle}>{text}</Text> : null;

  const buttonIcon = iconProps ? (
    <View style={iconStyle}>
      <Icon
        name={AppUtils.getIconForPlatform(iconProps.name)}
        size={iconProps.size || 30 * rem}
        color={iconProps.color}
      />
    </View>
  ) : null;

  let iconTextOrder = (
    <View style={viewStyle}>
      {iconProps && iconProps.rightSide ? (
        <>
          {buttonText}
          {buttonIcon}
        </>
      ) : (
        <>
          {buttonIcon}
          {buttonText}
        </>
      )}
    </View>
  );

  return (
    <TouchableOpacity style={tOpacityStyle} onPress={onPress}>
      {iconTextOrder}
    </TouchableOpacity>
  );
};

CustomButton.propTypes = {
  // eslint-disable-next-line react/no-unused-prop-types
  iconOrText: function(props, propName, componentName) {
    if (!props.iconProps?.name && !props.text) {
      return new Error(`Icon or text is required in ${componentName}`);
    }
  },
  iconProps: PropTypes.exact({
    name: PropTypes.string.isRequired,
    size: PropTypes.number,
    rightSide: PropTypes.bool,
    color: PropTypes.string
  }),
  text: PropTypes.string,
  tOpacityStyle: ViewPropTypes.style,
  viewStyle: ViewPropTypes.style,
  // eslint-disable-next-line react/forbid-prop-types
  textStyle: PropTypes.object,
  iconStyle: ViewPropTypes.style,
  onPress: PropTypes.func.isRequired
};

export default React.memo(CustomButton);
