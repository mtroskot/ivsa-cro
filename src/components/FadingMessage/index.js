import React, { useEffect, useRef } from 'react';
import { connect } from 'react-redux';
import { Animated, Text } from 'react-native';
import { StringUtils } from 'src/utils';
import { popupMesssagePropTypes } from 'src/constants/propTypes';
import styles from 'src/components/FadingMessage/styles';

const popupAnimation = new Animated.Value(0);
const PopupMesssage = props => {
  const mounted = useRef(false);

  useEffect(() => {
    if (!mounted.current) {
      mounted.current = true;
    } else {
      Animated.timing(popupAnimation, {
        toValue: 1,
        duration: 800,
        useNativeDriver: true
      }).start();
      setTimeout(() => {
        Animated.timing(popupAnimation, {
          toValue: 0,
          duration: 800,
          useNativeDriver: true
        }).start();
      }, 1500);
    }
  }, [props.popupMessage]);

  const { message, position } = props.popupMessage;
  const positionStyle = position === 'top' ? styles.topPosition : styles.bottomPosition;
  if (StringUtils.isEmpty(message)) {
    return null;
  }
  return (
    <Animated.View style={[styles.popupMessageView, positionStyle, { opacity: popupAnimation }]}>
      <Text style={styles.message}>{message}</Text>
    </Animated.View>
  );
};

PopupMesssage.propTypes = {
  popupMessage: popupMesssagePropTypes.isRequired
};
PopupMesssage.defaultProps = {
  position: 'top'
};
const mapStateToProps = state => ({
  popupMessage: state.ui.popupMessage
});
export default connect(
  mapStateToProps,
  null
)(React.memo(PopupMesssage));
