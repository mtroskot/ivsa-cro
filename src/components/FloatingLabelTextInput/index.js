import React, { useState } from 'react';
import PropTypes from 'prop-types';
import { Animated, TextInput, View } from 'react-native';
import styles from 'src/components/FloatingLabelTextInput/styles';
import { dimensions } from 'src/styles';
import HooksUtils from 'src/utils/HooksUtils';

const { rem } = dimensions;
const minOffset = -5 * rem;
const FloatingLabelTextInput = ({
  value,
  onChangeText,
  floatingLabel,
  textInputRef,
  textInputStyle,
  customContainerStyle,
  placeholder,
  ...restProps
}) => {
  const [focused, setFocused] = useState(false);
  const [floatAnim] = useState(new Animated.Value(minOffset));
  const [maxOffset, setMaxOffset] = useState(minOffset);
  // RN has bug when component re-mounts, animation value is reset to initial value
  const [animValueFix, setAnimValueFix] = useState(null);

  HooksUtils.useDidUpdate(() => {
    if (value.length === 0) {
      Animated.timing(floatAnim, {
        toValue: focused ? maxOffset : minOffset,
        duration: 250,
        useNativeDriver: true
      }).start();
    }
  }, [focused]);

  HooksUtils.useDidUpdate(() => {
    if (value.length === 0 && focused) {
      setAnimValueFix(maxOffset);
    } else {
      setAnimValueFix(null);
    }
  }, [value]);

  const handleFocus = () => {
    setFocused(true);
  };

  const handleBlur = () => {
    setAnimValueFix(null);
    setFocused(false);
  };

  return (
    <View
      onLayout={event => {
        const { height } = event.nativeEvent.layout;
        if (value.length === 0) {
          setMaxOffset(styles.floatingLabel.fontSize - height + 3);
        }
      }}
      style={[
        styles.container,
        customContainerStyle,
        { justifyContent: value.length === 0 && !placeholder ? 'flex-end' : 'flex-start' }
      ]}>
      {floatingLabel && (
        <Animated.Text
          style={
            value.length === 0 && !placeholder
              ? {
                  ...styles.floatingLabel,
                  transform: [{ translateY: animValueFix ? animValueFix : floatAnim }]
                }
              : { ...styles.floatingLabel }
          }>
          {floatingLabel}
        </Animated.Text>
      )}
      <TextInput
        value={value}
        onChangeText={onChangeText}
        placeholder={placeholder}
        underlineColorAndroid="transparent"
        ref={textInputRef}
        style={[styles.defaultInput, textInputStyle]}
        onFocus={handleFocus}
        onBlur={handleBlur}
        {...restProps}
      />
    </View>
  );
};

FloatingLabelTextInput.propTypes = {
  value: PropTypes.string.isRequired,
  onChangeText: PropTypes.func.isRequired,
  floatingLabel: PropTypes.string,
  placeholder: PropTypes.string,
  textInputRef: PropTypes.shape({ current: PropTypes.any }),
  textInputStyle: PropTypes.oneOfType([PropTypes.object, PropTypes.array]),
  customContainerStyle: PropTypes.oneOfType([PropTypes.object, PropTypes.array])
};

export default React.memo(FloatingLabelTextInput);
