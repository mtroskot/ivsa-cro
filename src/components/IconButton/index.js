import React from 'react';
import { TouchableOpacity, ViewPropTypes } from 'react-native';
import Icon from 'react-native-vector-icons/Ionicons';
import PropTypes from 'prop-types';
import { AppUtils } from 'src/utils';
import { dimensions } from 'src/styles';
const { rem } = dimensions;

const IconButton = ({ iconName, onPress, disabled, viewStyle, imageStyle }) => {
  return (
    <TouchableOpacity
      onPress={onPress}
      style={viewStyle}
      disabled={disabled}
      hitSlop={{ top: 20, left: 20, bottom: 20, right: 20 }}>
      <Icon name={AppUtils.getIconForPlatform(iconName)} style={imageStyle} size={30 * rem} />
    </TouchableOpacity>
  );
};

IconButton.propTypes = {
  iconName: PropTypes.string.isRequired,
  onPress: PropTypes.func.isRequired,
  disabled: PropTypes.bool,
  viewStyle: ViewPropTypes.style,
  imageStyle: PropTypes.oneOfType([PropTypes.object, PropTypes.array])
};

export default React.memo(IconButton);
