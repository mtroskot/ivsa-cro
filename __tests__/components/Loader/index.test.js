import React from 'react';
import { View } from 'react-native';
import { shallow } from 'enzyme';
import { Loader } from 'src/components';

describe('Loader wrapper', () => {
  it('Loader renders correctly without text', () => {
    const props = {
      size: 'small',
      color: 'black'
    };
    const wrapper = shallow(<Loader {...props} />);
    expect(wrapper.first().type()).toEqual(View);
    expect(wrapper.find('ActivityIndicator')).toHaveLength(1);
    expect(wrapper.find('ActivityIndicator').prop('size')).toEqual('small');
    expect(wrapper.find('ActivityIndicator').prop('color')).toEqual('black');
    expect(wrapper.find('Text')).toHaveLength(0);
    expect(wrapper).toMatchSnapshot();
  });
  it('Loader renders correctly with text', () => {
    const props = { text: 'text' };
    const wrapper = shallow(<Loader {...props} />);
    expect(wrapper.first().type()).toEqual(View);
    expect(wrapper.find('ActivityIndicator')).toHaveLength(1);
    expect(wrapper.find('ActivityIndicator').prop('size')).toEqual('large');
    expect(wrapper.find('ActivityIndicator').prop('color')).toEqual('#1e5ad4');
    expect(wrapper.find('Text')).toHaveLength(1);
    expect(wrapper.find('Text').props().children).toEqual(props.text);
    expect(wrapper).toMatchSnapshot();
  });
});
