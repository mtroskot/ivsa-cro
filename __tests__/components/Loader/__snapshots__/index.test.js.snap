// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loader wrapper Loader renders correctly with text 1`] = `
<View
  style={
    Array [
      Object {
        "flex": 1,
        "justifyContent": "center",
      },
      undefined,
    ]
  }
>
  <ActivityIndicator
    animating={true}
    color="#1e5ad4"
    hidesWhenStopped={true}
    size="large"
  />
  <Text
    style={
      Object {
        "textAlign": "center",
      }
    }
  >
    text
  </Text>
</View>
`;

exports[`Loader wrapper Loader renders correctly without text 1`] = `
<View
  style={
    Array [
      Object {
        "flex": 1,
        "justifyContent": "center",
      },
      undefined,
    ]
  }
>
  <ActivityIndicator
    animating={true}
    color="black"
    hidesWhenStopped={true}
    size="small"
  />
</View>
`;
